include/swift/AST/Attr.h

```diff
@@ -2145,19 +2145,29 @@ class TransposeAttr final
 /// has a different name).
 class HasAsyncAlternativeAttr final : public DeclAttribute  {
 public:
-  /// An optional name of the async alternative function, where the name of the
-  /// attributed function is used otherwise.
-  const DeclNameRef Name;
 
-  HasAsyncAlternativeAttr(DeclNameRef Name, SourceLoc AtLoc, SourceRange Range)
-    : DeclAttribute(DAK_HasAsyncAlternative, AtLoc, Range, false),
-      Name(Name) {}
+  /// Source location of the async function name in the attribute
+  const SourceLoc AsyncFunctionNameLoc;
 
-  HasAsyncAlternativeAttr(SourceLoc AtLoc, SourceRange Range)
-    : DeclAttribute(DAK_HasAsyncAlternative, AtLoc, Range, false) {}
+  /// DeclName of the async function in the attribute
+  const DeclNameRef AsyncFunctionName;
```

Comment:
Is the plan to find the function in a separate PR?

Reply:
Ah, you said you're still working on it 👍. Happy for it to be in this PR/another.

Reply:
Yeah, I'm still working on grabbing the function. I'm going to wrap it into the PR so I don't have to keep rev'ing the module version number. If this feature is blocking though, I can get this cleaned up and do it later though.

Reply:
Nope, not blocking. Leaving in this PR is okay with me.